pkg/config/config.go

```diff
@@ -624,6 +625,10 @@ type Project struct {
 	GiteaURLs GiteaURLs `yaml:"gitea_urls,omitempty"`
 }
 
+type GoMod struct {
+	Proxy bool `yaml:",omitempty"`
```

Comment:
One thing we will need to consider is config options for the proxy and sumdb. This effectively leads us to the user needing to be able to specifying various config options (that broadly follow https://golang.org/ref/mod#private-modules):

* `GOPROXY` - whilst we will want to use the `cmd/go` default of proxy.golang.org, we need to handle the fact that some users will have private modules and/or want to use a different proxy.
* `GOPRIVATE` 
* `GONOPROXY`
* `GONOSUMDB`
* `GOINSECURE`

Not sure where you think these best live?

Reply:
just added a `env` section to `gomod`. Hopefully that does the trick :)

Reply:
Does `goreleaser` use the environment it inherits when running `cmd/go` builds etc, or does it set an explicit environment based on `goreleaser` config?

Reply:
it inherits from env, and you can also set on the goreleaser config if you want to, so the final env would be the merge of both